src/silverlabnwb/signature.py

```diff
@@ -66,6 +99,18 @@ def ignore_attribute(self, path, attr):
         """
         self._ignore_attributes.append((re.compile(path + '$'), attr))
 
+    def set_cast_path(self, path, expected_type, corrected_type):
+        """Cast path (dataset or attribute) to corrected_type
+
+        Needed because some datasets and attributes are of different type on different platforms.
```

Comment:
Should we instead be specifying the number of bits to use on all platforms, rather than using a platform-dependent type when writing?

Probably not worth addressing in this PR though!

Reply:
Do you mean when writing the NWB file or the signature? If the former, I don't think we can... If the latter, I think this way offers at least a little "protection" if the schema/pynwb changes, whereas specifying the type for every platform will hide that change. Maybe that's what we want though...

Reply:
I meant when writing the NWB file. Bit worrying if pynwb doesn't let us control that, but at least it's not our fault if so, and then yes, this approach is probably best.

Reply:
I may be wrong, and perhaps we are creating different types implicitly on different environments. Can check this in the future.